modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/trait/TraitUtils.java

```diff
@@ -207,6 +230,18 @@ public static RewindabilityTrait rewindability(RelTraitSet traits) {
         return traits.getTrait(RewindabilityTraitDef.INSTANCE);
     }
 
+    /** */
+    public static CorrelationTrait correlation(RelNode rel) {
+        return rel instanceof IgniteRel
```

Comment:
looks like : ((IgniteRel)rel).correlation() and correlation(rel.getTraitSet()) are the same.

Reply:
Sure. Take a look at the utility methods for other traits.